llvm/test/CodeGen/SPIRV/llvm-intrinsics/constrained-convert.ll

```diff
@@ -0,0 +1,181 @@
+; RUN: llc -mtriple=spirv64-unknown-unknown %s -o - | FileCheck %s
+; RUN: %if spirv-tools %{ llc -mtriple=spirv64-unknown-unknown %s -o - -filetype=obj | spirv-val %}
+
+; CHECK-DAG: OpName %[[#sf:]] "conv"
+; CHECK-DAG: OpName %[[#sf1:]] "conv1"
+; CHECK-DAG: OpName %[[#sf2:]] "conv2"
+; CHECK-DAG: OpName %[[#sf3:]] "conv3"
+; CHECK-DAG: OpName %[[#sf4:]] "conv4"
+; CHECK-DAG: OpName %[[#uf1:]] "conv5"
+; CHECK-DAG: OpName %[[#uf2:]] "conv6"
+; CHECK-DAG: OpName %[[#uf3:]] "conv7"
+; CHECK-DAG: OpName %[[#uf4:]] "conv8"
+; CHECK-DAG: OpName %[[#uf5:]] "conv9"
+; CHECK-DAG: OpName %[[#fs1:]] "conv10"
+; CHECK-DAG: OpName %[[#fs2:]] "conv11"
+; CHECK-DAG: OpName %[[#fs3:]] "conv12"
+; CHECK-DAG: OpName %[[#fs4:]] "conv13"
+; CHECK-DAG: OpName %[[#fs5:]] "conv14"
+; CHECK-DAG: OpName %[[#fu1:]] "conv15"
+; CHECK-DAG: OpName %[[#fu2:]] "conv16"
+; CHECK-DAG: OpName %[[#fu3:]] "conv17"
+; CHECK-DAG: OpName %[[#fu4:]] "conv18"
+; CHECK-DAG: OpName %[[#fu5:]] "conv19"
+; CHECK-DAG: OpName %[[#fe1:]] "conv20"
+; CHECK-DAG: OpName %[[#fe2:]] "conv21"
+; CHECK-DAG: OpName %[[#ft1:]] "conv22"
+; CHECK-DAG: OpName %[[#ft2:]] "conv23"
+
+; CHECK-DAG: OpConvertSToF %[[#]] %[[#]]
+; CHECK-DAG: OpConvertSToF %[[#]] %[[#]]
+; CHECK-DAG: OpConvertSToF %[[#]] %[[#]]
+; CHECK-DAG: OpConvertSToF %[[#]] %[[#]]
+; CHECK-DAG: OpConvertSToF %[[#]] %[[#]]
+; CHECK-DAG: OpConvertUToF %[[#]] %[[#]]
+; CHECK-DAG: OpConvertUToF %[[#]] %[[#]]
+; CHECK-DAG: OpConvertUToF %[[#]] %[[#]]
+; CHECK-DAG: OpConvertUToF %[[#]] %[[#]]
+; CHECK-DAG: OpConvertUToF %[[#]] %[[#]]
+; CHECK-DAG: OpConvertFToS %[[#]] %[[#]]
+; CHECK-DAG: OpConvertFToS %[[#]] %[[#]]
+; CHECK-DAG: OpConvertFToS %[[#]] %[[#]]
+; CHECK-DAG: OpConvertFToS %[[#]] %[[#]]
+; CHECK-DAG: OpConvertFToS %[[#]] %[[#]]
+; CHECK-DAG: OpFConvert %[[#]] %[[#]]
+; CHECK-DAG: OpFConvert %[[#]] %[[#]]
+; CHECK-DAG: OpFConvert %[[#]] %[[#]]
+; CHECK-DAG: OpFConvert %[[#]] %[[#]]
+
+; CHECK-DAG: OpDecorate %[[#sf]] FPRoundingMode RTE
+; CHECK-DAG: OpDecorate %[[#sf1]] FPRoundingMode RTZ
+; CHECK-DAG: OpDecorate %[[#sf2]] FPRoundingMode RTP
+; CHECK-DAG: OpDecorate %[[#sf3]] FPRoundingMode RTN
+; CHECK-DAG: OpDecorate %[[#sf4]] FPRoundingMode RTE
+; CHECK-DAG: OpDecorate %[[#uf1]] FPRoundingMode RTE
+; CHECK-DAG: OpDecorate %[[#uf2]] FPRoundingMode RTZ
+; CHECK-DAG: OpDecorate %[[#uf3]] FPRoundingMode RTP
+; CHECK-DAG: OpDecorate %[[#uf4]] FPRoundingMode RTN
+; CHECK-DAG: OpDecorate %[[#uf5]] FPRoundingMode RTE
+; CHECK-DAG: OpDecorate %[[#ft1]] FPRoundingMode RTZ
+; CHECK-DAG: OpDecorate %[[#ft2]] FPRoundingMode RTE
+
+define dso_local spir_kernel void @test1(i32 %in) {
```

Comment:
Don't need dso_local, Also would be better to use a non-kernel and have a real use of the return value